src/patch.js

```diff
@@ -50,6 +51,12 @@ function updateComponent (oldVirtualNode, newVirtualNode, options) {
       delete newProps.ref
     }
   }
+
+  if (newProps && newProps.key) {
```

Comment:
I'm somewhat worried about the overhead of always copying the props object. I deliberately avoided doing it in other places such as populated the `events` sub-object. Thinking about our alternatives...

* We could mutate the props object we are given. This is somewhat more dangerous than the events case because we're losing information at that point and if the props object gets reused somehow it could be trouble.
* We could only clone the object and delete the key when passing props down to a *component*, then explicitly ignore the key in our normal DOM patching operations.
* We could change the logic and just pass the key through to the component. It would be the component's responsibility to ignore it. This seems maybe the simplest and fastest to me but could require changes to existing application code using previous Etch versions.